scalafix-core/shared/src/main/scala/scalafix/internal/rule/OrphanImplicits.scala

```diff
@@ -0,0 +1,66 @@
+package scalafix.internal.rule
+
+import scala.meta._
+import scalafix.internal.util.SymbolOps
+import scalafix.lint.LintCategory
+import scalafix.rule.RuleCtx
+import scalafix.util.SemanticdbIndex
+import scalafix.{LintMessage, Rule}
+
+final case class OrphanImplicits(index: SemanticdbIndex)
+    extends Rule("OrphanImplicits") {
+
+  override def description: String =
+    "Linter that reports an error on implicit instances of shape F[G] " +
+      "that do not belong in the companion objects of F or G."
+
+  private lazy val errorCategory: LintCategory =
+    LintCategory.error("Orphan implicits should be avoided")
+
+  override def check(ctx: RuleCtx): Seq[LintMessage] = {
+
+    def handleImplicit(
+        tpe: Type,
+        obj: Symbol.Global,
+        pos: Position): Option[LintMessage] = {
+
+      val tpeSymbols = tpe match {
+        case Type.Apply(t, Seq(arg)) =>
```

Comment:
This won't handle cases with multiple type arguments

```scala
object Foo {
  implicit val x: Map[String, Int] = Map.empty
}
```

Reply:
Yeah, it won't :) As I understand @fommil wants only F[G] form or some complicated detection of fixed type-vars.

Reply:
in hindsight, #661 is a prerequisite for doing orphan implicits properly... it bans everything except typeclasses. And then with what's left we can seach on all the type parameters to see if it's defined in a good place or not.